openwisp_firmware_upgrader/base/models.py

```diff
@@ -124,10 +125,17 @@ def _find_firmwareless_devices(self, boards=None):
                                      model__in=boards)
 
 
+def get_build_directory(instance, filename):
+    build_id = str(instance.build.id)
```

Comment:
use `pk` because developers extending this module may replace the uuid with a different kind of primary key (it happened once in django-x509).